harness/determined/common/experimental/model.py

```diff
@@ -218,18 +218,13 @@ def get_versions(self, order_by: ModelOrderBy = ModelOrderBy.DESC) -> List[Model
             order_by (enum): A member of the :class:`ModelOrderBy` enum.
         """
 
-        def get_with_offset(offset: int) -> bindings.v1GetModelVersionsResponse:
-            return bindings.get_GetModelVersions(
-                self._session,
-                modelName=self.name,
-                orderBy=order_by._to_bindings(),
-            )
-
-        resps = api.read_paginated(get_with_offset)
```

Comment:
This is a paginated endpoint, and `read_paginated` is designed to abstract that away.

It is true that setting limit=0 makes the endpoint effectively not paginated in today's implementation, but it is also true that if backend decides to change that behavior some day, the proposed code would break but the `read_paginated()` version would not break.

So I'd say: "paginated endpoint means use read_paginated, and no further consideration is necessary".

Reply:
You're right. I'll fix.

Reply:
I forced push and amended commit thinking that might change the commit message in GitHub, too. That didn't happen, so instead I've updated it manually.

Thanks for asking for the change. This is good, defensive programming. I'd wanted to simplify the code, but this is clearly the right choice to make, instead.